pkg/utils/filesystem.go

```diff
@@ -83,6 +86,24 @@ func WriteToComposeFile(dockerComposeFile string, debug bool) bool {
 	return true
 }
 
+func writeDockerConfigSecretFile() {
+	dockerConfig := getDockerCredentials("local")
+	dockerConfigBytes, jsonErr := json.MarshalIndent(dockerConfig, "", "  ")
+	if jsonErr != nil {
+		errors.CheckErr(jsonErr, 208, "")
+	}
+	encoded := base64.StdEncoding.EncodeToString(dockerConfigBytes)
+	err := ioutil.WriteFile(dockerConfigSecretFile, []byte(encoded), 0600)
+	errors.CheckErr(err, 204, "")
```

Comment:
`writeDockerConfigSecretFile` was only called in `filesystem.go`, so I've pushed the error handling up one level there. (We can fix that use of CheckErr when we fix the rest of `WriteToComposeFile`.)
`ClearDockerConfigSecret` now returns an error *but* as it's called in the cleanup path it mostly won't be checked, we don't want a failure to erase the secret to stop us shutting down or cleaning up another error. Technically that is at least the callers decision, calling `CheckErr` which then went on to use `log.Fatal` was probably the wrong thing to do as it could have messed up shutdown if we couldn't write to the secret file.